diagnostic.md

```diff
@@ -73,20 +98,49 @@ data in JSON format.
 ```
 
 ```sh
-# your answer here
+curl -- request POST http://www.example.com/donuts/ \
+--header "Content-Type: application/json"
+--data '{
+  "name": "French Cruller",
+  "price": "$0.99"
+}'
 ```
 
 Write an AJAX request to create a single donut on the server using JSON.
 
 ```js
-let createDonut = /* your answer here */;
+let createDonut =  function(event) {
+event.preventDefault();
+url: 'http://www.example.com/donuts/15'
+type: 'POST'
+data: {
+  'name': 'French Cruller',
+  'price': '$0.99'
+  'id': '18'
+}
+})
+  .done(logResponseBody)
+  .fail(logRequestError);
+};
 ```
 
 Now, rewrite your AJAX request to use FormData. Assume you have already targeted
 the correct HTML form before registering your AJAX function with the form.
 
 ```js
-let createDonut = /* your answer here */;
+let createDonut = function(event) {
+event.preventDefault();
+url: 'http://www.example.com/donuts/15'
+type: 'POST'
+data: {
+  'name': $(event.target).find('#name').val(),
+  'price': $(event.target).find('#price').val(),
+  'id': $(event.target).find('#id').val(),
+}
```

Comment:
The point of using `FormData` is so that we don't have to do precisely this. 
We can write
```
let donut = new FormData(event.target);
$.ajax({
  url: 'http://www.example.com/donuts',
  type: 'POST',
  data: donut,
  contentType: false,
  processData: false
})
.done(logResponseBody)
.fail(logRequestError);
```

FormData will handle reading from all the form fields on the page.